### PR TITLE
admin: Add /reopen_logs admin handler

### LINKED ISSUE #10266: /reopen_logs admin handler
In our environment processes run in isolated process namespaces and cannot send signals to each other.
We need to implement correct log rotation for envoy, and possible solution is to expose /reopen_logs admin handler which triggers AccessLogManager's reopen() similar to SIGUSR1 handler.

Implementation might be pretty straightforward:
```
Http::Code AdminImpl::handlerReopenLogs(absl::string_view, Http::HeaderMap&,
                                          Buffer::Instance& response, AdminStream&) {
  server_.accessLogManager().reopen();
  response.add("OK\n");
  return Http::Code::OK;
}
```

### PR DESCRIPTION
**Pull request by veshij**
This patch adds `/reopen_logs` admin handler to control log rotation.

Risk Level: Low
Testing: Unit test
Docs Changes: Added
Release Notes: Added
Fixes #10266

Signed-off-by: Oleg Guba <oleg@dropbox.com>

### CODE REVIEW

**Comment by mattklein123** [comment]
At a high level LGTM, thanks. Obviously needs release notes, docs, etc. Thank you!

/wait

File: source/extensions/filters/network/common/redis/codec_impl.cc [hunk 1]
```diff
@@ -316,8 +316,8 @@ RespValue::CompositeArray::CompositeArrayConstIterator::operator++() {
   return *this;
 }
 
-bool RespValue::CompositeArray::CompositeArrayConstIterator::operator!=(
-    const CompositeArrayConstIterator& rhs) const {
+bool RespValue::CompositeArray::CompositeArrayConstIterator::
+operator!=(const CompositeArrayConstIterator& rhs) const {
```
Comment (dio) [inline source/extensions/filters/network/common/redis/codec_impl.cc]: I think this change is not necessary? This is causing the format error.

Comment (veshij) [reply source/extensions/filters/network/common/redis/codec_impl.cc]: Sorry for delay, was on PTO.
This change was done by one of the linters as without it other test was failing.
Let me try to run all linters again.

Comment (veshij) [reply source/extensions/filters/network/common/redis/codec_impl.cc]: /wait

**Comment by dio** [comment]
@veshij as noted by Matt, could you add release notes (docs/root/intro/version_history.rst) and relevant docs? Thank you!

File: docs/root/operations/admin.rst [hunk 2]
```diff
@@ -558,3 +558,7 @@ modify different aspects of the server:
   been configured to accept admin configuration. See:
 
   * :ref:`HTTP tap filter configuration <config_http_filters_tap_admin_handler>`
+
+.. http:get:: /reopen_logs
```
Comment (mattklein123) [inline docs/root/operations/admin.rst]: Sorry that I didn't get this before, but I think this should be a POST as it changes state. cc @jmarantz

File: source/server/http/admin.cc [hunk 3]
```diff
@@ -1449,6 +1456,8 @@ AdminImpl::AdminImpl(const std::string& profile_path, Server::Instance& server)
           {"/runtime", "print runtime values", MAKE_ADMIN_HANDLER(handlerRuntime), false, false},
           {"/runtime_modify", "modify runtime values", MAKE_ADMIN_HANDLER(handlerRuntimeModify),
            false, true},
+          {"/reopen_logs", "reopen access logs", MAKE_ADMIN_HANDLER(handlerReopenLogs), false,
+           false},
```
Comment (mattklein123) [inline source/server/http/admin.cc]: Per my above comment I think you want a true param here to make this a POST?

File: docs/root/operations/admin.rst (hunk 2 above)
Comment (veshij) [reply docs/root/operations/admin.rst]: np, will update today.

Comment (jmarantz) [reply docs/root/operations/admin.rst]: agreed.

File: test/server/http/admin_test.cc [hunk 4]
```diff
@@ -1249,6 +1249,16 @@ TEST_P(AdminInstanceTest, RuntimeModifyNoArguments) {
...
 }
 
+TEST_P(AdminInstanceTest, ReopenLogs) {
+  Http::ResponseHeaderMapImpl header_map;
+  Buffer::OwnedImpl response;
+  testing::NiceMock<AccessLog::MockAccessLogManager> access_log_manager_;
+
+  EXPECT_CALL(server_, accessLogManager()).WillRepeatedly(ReturnRef(access_log_manager_));
+  EXPECT_CALL(access_log_manager_, reopen());
+  EXPECT_EQ(Http::Code::OK, getCallback("/reopen_logs", header_map, response));
```
Comment (mattklein123) [inline test/server/http/admin_test.cc]: I think you want `postCallback` here?

/wait

**Pull request merged**